src/internal/data/data_collection.go

```diff
@@ -47,6 +48,11 @@ type StreamSize interface {
 	Size() int64
 }
 
+// StreamModTime is used to provide the modified time of the stream's data.
+type StreamModTime interface {
```

Comment:
Nit: slightly more in keeping with go interface naming convention.
```suggestion
// ModTimer is used to provide the modified time of the stream's data.
type ModTimer interface {
```

Reply:
hmmm actually now that I look at it more, can we hold off on this. I wanted to keep it consistent with some of the other interfaces defined in the file (`StreamSize`, `StreamInfo`). I think either we should change all of them in a different patch or leave them all consistency